### PR TITLE
Remove empty lines and make kotlin null safety handling more elegant

### PR DESCRIPTION
**Pull request by curtishd**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [x] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [x] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [x] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

File: codes/kotlin/chapter_tree/binary_search_tree.kt [hunk 1]
```diff
@@ -112,13 +110,13 @@ class BinarySearchTree {
         } else {
             // 获取中序遍历中 cur 的下一个节点
             var tmp = cur.right
-            while (tmp!!.left != null) {
-                tmp = tmp.left
+            tmp?.left?.let {
```
Comment (krahets) [inline codes/kotlin/chapter_tree/binary_search_tree.kt:113]: Is it equivalant?

File: codes/kotlin/chapter_tree/avl_tree.kt [hunk 2]
```diff
@@ -144,11 +143,11 @@ class AVLTree {
             } else {
                 // 子节点数量 = 2 ，则将中序遍历的下个节点删除，并用该节点替换当前节点
                 var temp = node.right
-                while (temp!!.left != null) {
-                    temp = temp.left
+                temp?.left?.let {
```
Comment (krahets) [inline codes/kotlin/chapter_tree/avl_tree.kt:146]: Is it equivalent?

Comment (curtishd) [reply codes/kotlin/chapter_tree/avl_tree.kt:146]: The output is same, In this situation.

Comment (curtishd) [reply codes/kotlin/chapter_tree/avl_tree.kt:146]: Maybe is inappropriate, I will change it back.

Comment (krahets) [reply codes/kotlin/chapter_tree/avl_tree.kt:146]: The same outputs do not mean they're equivalent. There is no `while` loop in the new code.

File: codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt [hunk 3]
```diff
@@ -82,8 +82,8 @@ class LinkedListDeque {
             _val = rear!!._val // 暂存尾节点值
             // 删除尾节点
             val rPrev = rear!!.prev
-            if (rPrev != null) {
-                rPrev.next = null
+            rPrev?.let {
```
Comment (krahets) [inline codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt:85]: The readability is worse. Recommend revert it.

File: codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt [hunk 4]
```diff
@@ -64,16 +64,16 @@ class LinkedListDeque {
...
             throw IndexOutOfBoundsException()
         val _val: Int
         // 队首出队操作
         if (isFront) {
             _val = front!!._val // 暂存头节点值
             // 删除头节点
             val fNext = front!!.next
-            if (fNext != null) {
-                fNext.prev = null
+            fNext?.let {
```
Comment (krahets) [inline codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt:75]: Is it equivalent?

File: codes/kotlin/chapter_tree/avl_tree.kt (hunk 2 above)
Comment (curtishd) [reply codes/kotlin/chapter_tree/avl_tree.kt]: I confirm, it`s been changed.

File: codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt (hunk 4 above)
Comment (krahets) [reply codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt:75]: I think `fNext?.let` means `if (fNext == null) then`

File: codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt (hunk 3 above)
Comment (krahets) [inline codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt:85]: Same as above

File: codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt (hunk 4 above)
Comment (curtishd) [reply codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt:75]: yes, it is equivalent.

Comment (curtishd) [reply codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt:75]: fNext ?:  means if (fNext == null) then

File: codes/kotlin/chapter_dynamic_programming/min_cost_climbing_stairs_dp.kt [hunk 5]
```diff
@@ -31,9 +31,7 @@ fun minCostClimbingStairsDPComp(cost: IntArray): Int {
     var a = cost[1]
     var b = cost[2]
     for (i in 3..n) {
-        val tmp = b
-        b = min(a, tmp) + cost[i]
-        a = tmp
+        a = b.also { b = min(a, b) + cost[i] }
```
Comment (krahets) [inline codes/kotlin/chapter_dynamic_programming/min_cost_climbing_stairs_dp.kt:34]: The readability is not good

File: codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt (hunk 4 above)
Comment (krahets) [reply codes/kotlin/chapter_stack_and_queue/linkedlist_deque.kt:75]: Which one is correct:

1. `if (fNext == null)` euqivalent to `fNext?.let {`
2. `if (fNext != null)` euqivalent to `fNext?.let {`

**Pull request closed without merging**